src/cloudai/schema/system/slurm/slurm_system.py

```diff
@@ -493,61 +493,99 @@ def scancel(self, job_id: int) -> None:
         """
         self.cmd_shell.execute(f"scancel {job_id}")
 
-    def update_node_states(self) -> None:  # noqa: C901
+    def update_node_states(self, squeue_output: Optional[str] = None, sinfo_output: Optional[str] = None) -> None:
```

Comment:
I suggest we make arguments non-optional. Caller should prepare it in advance. This simplifies the function and the way we test it.

Reply:
I introduced these primarily for testing the update_node_states function. My intention is to keep the interface simple. When you examine the caller sites of this function, you'll see it is used to ensure all node states are up-to-date. Mandating users to pass squeue and sinfo would complicate the calling process.

Reply:
Now you can mock `get_squeue()` and `get_sinfo()` outputs and remove this args to keep this interface simple.

Reply:
I see, I will remove the args then :)

Reply:
Updated.